.github/workflows/ci-tk.yaml

```diff
@@ -22,7 +22,90 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event.number || github.sha }}
   cancel-in-progress: true
 
+env:
+  LLVM_SHA_FILE: llvm-sha.txt
+  LLVM_CACHE_NUMBER: 1  # Increase to reset cache
+  WATER_REF: 1a84aa14ee74d5015ceb41bed438de4389f59b83
+
 jobs:
+  build_llvm_linux:
```

Comment:
Can you add a comment here explaining why LLVM needs to be built from source, perhaps discussing alternatives to consider in the future?

Reply:
I'd love to be able to just link to libMLIR.so and have iree do the same, but I don't think that was ever made to work across platforms.

Reply:
Please no, dynamic LLVM/MLIR linking is huge footgun and I would prefer for it to not exist :)

Reply:
added the comment